### PR TITLE
Convert all code to use the ralith branch of ruma-api.

### PR DESCRIPTION
**Pull request by jimmycuadra**
This is a work in progress to use the version of ruma-api proposed in https://github.com/ruma/ruma-api/pull/9, so we can see how well it actually works. I want to try integrating this branch of ruma-client-api into both ruma-client and ruma, too, as that is the _real_ test of how well it fits. Feedback welcome!

/cc @Ralith @vberger @eternaleye @jplatte @exul @neosam

### CODE REVIEW

File: src/unversioned.rs [hunk 1]
```diff
@@ -2,49 +2,81 @@
...
-        fn name() -> &'static str {
-            "api_version"
-        }
+    impl TryFrom<::Request> for Request {
+        type Err = ::Error;
 
-        fn description() -> &'static str {
-            "Get the versions of the client-server API supported by this homeserver."
+        fn try_from(_request: ::Request) -> Result<Self, Self::Err> {
+            Ok(Request)
```
Comment (vberger) [inline src/unversioned.rs]: Question: Should this return an error if superfluous parameters are given?

**Comment by vberger** [comment]
I like the way it looks overall 👍 

I still think that a good part of the endpoint definition could be factored with a macro, I'll think about that when I get some time.

File: src/unversioned.rs (hunk 1 above)
Comment (jimmycuadra) [inline src/unversioned.rs]: I'd say not.

**Comment by jimmycuadra** [comment]
I converted the `account` module, which provides a more substantial example than `unversioned`. I'm curious what everyone thinks.

**Comment by jimmycuadra** [comment]
Converted the `alias` module too. At this point I'm gonna experiment with a macro to reduce boilerplate, as the implementation patterns are becoming clearer.

**Comment by neosam** [comment]
The contact module looks a lot cleaner with the macros.  For me it feels a lot easier to get started with the new code.

**Comment by jimmycuadra** [comment]
Closing in favor of the new ruma-api API. Manually implemented the `unversioned` module on the "manual" branch, but my plan is to have all of these endpoints generated through a Macros 2.0 procedural macro, which will likely be an evolution of [ruma-api-derive](https://github.com/ruma/ruma-api-derive).

**Pull request closed without merging**